hazelcast/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -325,6 +332,8 @@ public void run() {
         // prevent lambda allocation on each iteration
         private final Consumer<TaskletTracker> runTasklet = this::runTasklet;
 
+        private final Semaphore hasTasklet = new Semaphore(1);
```

Comment:
Doesn't really change anything, but why we start with 1 permit? Should be 0.
A better name might be `newTaskletsSemaphore`.